Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -219,6 +219,12 @@ public void reheap(boolean validate) {
         }
     }
 
+    public void addPoint(Integer pointIndex, float weight, long sequenceIndex) {
```

Comment:
Why did this method get added now? What changed?

Reply:
This is to enable double[] -> float[] conversion for the trees. Up until now we were bringing up the exact same samplers in ser-de. The guarantee we maintained is that for duplicates in the same tree, the corresponding sampler would have the same pointIndex value. In the conversion, we first round off the pointstore -- but then we may be creating duplicates. So we have to insert the points into a fresh tree and update the pointIndex. One option is to add the point to the samplerPlusTree and update both sampler and tree -- we could use the existing code. Other option is to add them to tree but not affect the sampler (other than updating the pointIndex value) -- which is happening here. Advantage of option 2 is that the sampler effectively remains in the same state; the scores seem a bit smoother in the conversion. Looking far down the road, this should also enable adding "sticky"  points to a sampler -- but that would require a resize() for grow() option to dynamically change (increase is the difficult part) the sampler size.